packages/deck/backend/sirius-components-collaborative-deck/src/main/java/org/eclipse/sirius/components/collaborative/deck/service/DeckCardService.java

```diff
@@ -0,0 +1,196 @@
+
+        if (optionalCard.isPresent() && optionalCardDescription.isPresent()) {
+            Optional<Object> optionalTargetObject = this.objectService.getObject(editingContext, optionalCard.get().targetObjectId());
+            if (optionalTargetObject.isPresent()) {
+                VariableManager variableManager = new VariableManager();
+                variableManager.put(VariableManager.SELF, optionalTargetObject.get());
+                optionalCardDescription.get().deleteCardProvider().accept(variableManager);
+
+                payload = this.getPayload(deleteDeckCardInput.id());
+            }
+        }
+        return payload;
+    }
+
+    @Override
+    public IPayload editCard(EditDeckCardInput editDeckCardInput, IEditingContext editingContext, Deck deck) {
+        IPayload payload = new ErrorPayload(editDeckCardInput.id(), "Edit card failed");
+
+        Optional<Card> optionalCard = this.findCard(card -> Objects.equals(card.id(), editDeckCardInput.cardId()), deck);
+        Optional<CardDescription> optionalCardDescription = optionalCard.flatMap(card -> this.findCardDescription(card.descriptionId(), deck, editingContext));
+
+        if (optionalCard.isPresent() && optionalCardDescription.isPresent()) {
+            Optional<Object> optionalTargetObject = this.objectService.getObject(editingContext, optionalCard.get().targetObjectId());
+            if (optionalTargetObject.isPresent()) {
+                VariableManager variableManager = new VariableManager();
+                variableManager.put(VariableManager.SELF, optionalTargetObject.get());
+                variableManager.put(DeckDescription.NEW_TITLE, editDeckCardInput.newTitle());
+                variableManager.put(DeckDescription.NEW_DESCRIPTION, editDeckCardInput.newDescription());
+                variableManager.put(DeckDescription.NEW_LABEL, editDeckCardInput.newLabel());
+                optionalCardDescription.get().editCardProvider().accept(variableManager);
+
+                payload = this.getPayload(editDeckCardInput.id());
+            }
+        }
+
+        return payload;
+    }
+
+    private IPayload getPayload(UUID payloadId) {
+        IPayload payload = null;
+        List<Message> feedbackMessages = this.feedbackMessageService.getFeedbackMessages();
+        Optional<Message> optionalErrorMsg = feedbackMessages.stream().filter(msg -> MessageLevel.ERROR.equals(msg.level())).findFirst();
+        if (optionalErrorMsg.isPresent()) {
+            payload = new ErrorPayload(payloadId, optionalErrorMsg.get().body(), feedbackMessages);
+        } else {
+            payload = new SuccessPayload(payloadId, feedbackMessages);
+        }
+        return payload;
+    }
+    private Optional<Lane> findLane(Predicate<Lane> condition, Deck deck) {
+        return deck.lanes().stream()
+                .filter(condition::test)
```

Comment:
condition is already a predicate, you don't have to call test you could just use `.filter(condition)`

Reply:
done